controllers/operands/networkAddons.go

```diff
@@ -149,6 +150,12 @@ func NewNetworkAddons(hc *hcov1beta1.HyperConverged, opts ...string) (*networkad
 		if baseDomain != "" {
 			cnaoSpec.KubeSecondaryDNS.Domain = baseDomain
 		}
+		if nameServerIP := *hc.Spec.KubeSecondaryDNSNameServerIP; nameServerIP != "" {
```

Comment:
Is it working when the field is nil? I think it will cause segmentation violation error. See here for example: https://go.dev/play/p/UjC-oXlIn-e

I think we must check for nil before accessing this field.